src/gammaln.js

```diff
@@ -0,0 +1,58 @@
+/* @flow */
+
+/**
+ * Compute the logarithm of the [gamma function](https://en.wikipedia.org/wiki/Gamma_function) of a value using Lanczos' approximation.
+ * This function takes as input any real-value n greater than 0.
+ * This function is useful for values of n too large for the normal gamma function (n > 165).
+ * The code is based on Lanczo's Gamma approximation, defined [here](http://my.fit.edu/~gabdo/gamma.txt).
+ *
+ * @param {number} n Any real number greater than zero.
+ * @returns {number} The logaritm of gamma of the input value.
+ *
+ * @example
+ * gammaln(500); // 2605.1158503617335
+ * gammaln(2.4); // 0.21685932244884043 
+ */
+function gammaln(n /*: number */ ) /*: number */ {
+
+    // Return infinity if value not in domain
+    if (n <= 0) {
+        return Infinity;
+    }
+    
+    // Decrement n, because approximation is defined for n - 1
+    n--;
+    
+    // Define series coefficients
+    var coefficients = [
```

Comment:
Since they are static, you should define those coefficients outside of the function so you don't create a new array each time the function is called.